### PR TITLE
[Internal] Improve experience in PR reviews #trivial

### PR DESCRIPTION
**Pull request by AliSoftware**
* [x] I've started my branch from the `develop` branch (gitflow)
  * [x] And I've selected `develop` as the "base" branch for this Pull Request I'm about to create
* [x] I've added an entry in the `CHANGELOG.md` file to explain my changes and credit myself  
      _(or added `#trivial` to the PR title if I think it doesn't warrant a mention in the CHANGELOG)_
  * [ ] Add the entry in the appropriate section (Bug Fix / New Feature / Breaking Change / Internal Change)
  * [ ] Add a period and 2 spaces at the end of your short entry description
  * [ ] Add links to your GitHub profile to credit yourself and to the related PR and issue after your description

---

* Don't hide `compilation-configuration.yml` files in PRs
* Detect common typos in GitHub links to issues/PRs in CHANGELOG

It looks like this:
image

### CODE REVIEW

**Comment by SwiftGen-Eve** [comment]
Hey 👋 I'm Eve, the friendly bot watching over SwiftGen 🤖

Thanks a lot for your contribution!



---

Seems like everything is in order 👍 You did a good job here! 🤝

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: CHANGELOG.md [hunk 1]
```diff
@@ -57,7 +57,7 @@ There are no major changes in this release, although JSON & Plist template write
 * Avoid breaking the system swift installation when installing SwiftGen via Homebrew on macOS 10.14.4 or higher.  
   [David Jennes](https://github.com/djbe)
   [#686](https://github.com/SwiftGen/SwiftGen/issue/686)
```
Comment (SwiftGen-Eve) [inline CHANGELOG.md]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/SwiftGen/SwiftGen/blob/96c53d24240cdcf7aa236557505ede9b1a207372/CHANGELOG.md#L59"></span>This looks like a GitHub link URL with a typo. Issue links should use <code>/issues/123</code> (plural) and PRs links should use <code>/pull/123</code> (singular).</td>
    </tr>
  </tbody>
</table>

File: CHANGELOG.md [hunk 2]
```diff
@@ -57,7 +57,7 @@ There are no major changes in this release, although JSON & Plist template write
...
-  [#730](https://github.com/SwiftGen/SwiftGen/pull/730)
+  [#730](https://github.com/SwiftGen/SwiftGen/pulls/730)
```
Comment (SwiftGen-Eve) [inline CHANGELOG.md]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/SwiftGen/SwiftGen/blob/96c53d24240cdcf7aa236557505ede9b1a207372/CHANGELOG.md#L60"></span>This looks like a GitHub link URL with a typo. Issue links should use <code>/issues/123</code> (plural) and PRs links should use <code>/pull/123</code> (singular).</td>
    </tr>
  </tbody>
</table>

**Review by djbe — APPROVED** [review]

File: rakelib/check_changelog.rb [hunk 3]
```diff
@@ -45,6 +48,12 @@ def check_changelog
       link_url = "#{m[2]}##{m[4]}"
       { line: idx+1, message: "Link text is #{link_text} but links points to #{link_url}." }
     end)
+
+    # Flag common typos in GitHub issue/PR URLs
+    typo_links = line.scan(links_typos)
+    all_warnings.concat Array(typo_links.map do |m|
+      { line: idx+1, message: "This looks like a GitHub link URL with a typo. Issue links should use `/issues/123` (plural) and PRs links should use `/pull/123` (singular)." }
```
Comment (djbe) [inline rakelib/check_changelog.rb]:
```suggestion
      { line: idx+1, message: "This looks like a GitHub link URL with a typo. Issue links should use `/issues/123` (plural) and PR links should use `/pull/123` (singular)." }
```

**Review by djbe — CHANGES_REQUESTED** [review]
Nit

File: rakelib/check_changelog.rb (hunk 3 above)
Comment (AliSoftware) [reply rakelib/check_changelog.rb]: You mean I made a typo in the code that warns about typos? 😆

Comment (AliSoftware) [reply rakelib/check_changelog.rb]: fixed.

**Review by djbe — APPROVED** [review]

**Pull request merged**